### PR TITLE
Workbench certs

### PR DESCRIPTION
**Pull request by KPostOffice**
# Issue link
supersedes #489 

# What changes have been made
I made a minor change to #489 since @Bobbins228 is out. This change makes sure the explicitly passed arg takes priority over the default

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/cluster/auth.py [hunk 1]
```diff
@@ -106,10 +118,20 @@ def login(self) -> str:
             configuration.api_key_prefix["authorization"] = "Bearer"
             configuration.host = self.server
             configuration.api_key["authorization"] = self.token
-            if self.skip_tls == False and self.ca_cert_path == None:
+
+            if self.skip_tls == False:
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk/cluster/auth.py:122]:
```suggestion
            if not self.skip_tls:
```
Nitpick for a more Pythonic style.

File: src/codeflare_sdk/cluster/auth.py [hunk 2]
```diff
@@ -106,10 +118,20 @@ def login(self) -> str:
...
+                if self.ca_cert_path == None:
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk/cluster/auth.py:123]:
```suggestion
                if self.ca_cert_path is None:
```

File: tests/unit_test.py [hunk 3]
```diff
@@ -131,19 +131,24 @@ def test_token_auth_creation():
...
         assert token_auth.ca_cert_path == None
 
+        os.environ["CF_SDK_CA_CERT_PATH"] = f"/etc/pki/tls/custom-certs/ca-bundle.crt"
         token_auth = TokenAuthentication(token="token", server="server", skip_tls=False)
         assert token_auth.token == "token"
         assert token_auth.server == "server"
         assert token_auth.skip_tls == False
-        assert token_auth.ca_cert_path == None
+        assert token_auth.ca_cert_path == "/etc/pki/tls/custom-certs/ca-bundle.crt"
+        os.environ.pop("CF_SDK_CA_CERT_PATH")
```
Comment (ChristianZaccaria) [inline tests/unit_test.py:140]: I think this line is not necessary, further proving that priority is given to the `ca_cert_path` parameter even if the environment variable is set.

**Review by ChristianZaccaria — CHANGES_REQUESTED** [review]
Tested:
- Priority is given to the `ca_cert_path` argument over the environment variable set.
- Can authenticate with certificate located through the path specified by the environment variable.
- Ran through a notebook and submitted job successfully.

Thank you, just left a few nitpicks.

File: tests/unit_test.py (hunk 3 above)
Comment (KPostOffice) [reply tests/unit_test.py:140]: It was failing some other unit tests since they don't expect the env var to be set

**Review by ChristianZaccaria — APPROVED** [review]

**Pull request merged**